### ISSUE TITLE
RuntimeError: time-out of 2000ms while waiting for the simulator, make sure the simulator is ready and connected to 127.0.0.1:2000

### ISSUE THREAD

**Issue by akshatamk** [carla-simulator/carla#3430]
Hello,
   I'm getting this error while running the python API in CARLA.
   I tried changing time-out duration as well as port number, but still the same issue. Please help

**Comment by tinmodeHuang** [comment]
just now I have ran into same error and had no idea how to solve it as well, however, I can perform spawn_npc.py successfully after changing into newest edition

**Comment by jackbart94** [comment]
Hi,

Could you tell us more about this, for example the steps you are taking to execute CARLA? Are you running it in Unreal Engine, or in the package version?

**Comment by akshatamk** [comment]
Thanks for the response.
yes i'm running CARLA in unreal engine 4.24x.The simulator is opening but unable to run the example APIs.

**Comment by jackbart94** [comment]
Did you press the Play button (or alternatively Alt+P) after launching Unreal Engine, before running the scripts? Because I've only seen this error when you forget to do so.

**Comment by akshatamk** [comment]
Currently I'm trying to set up CARLA on azure Windows VM with (NC6s_V3-Tesla K80), does this environment support Unreal engine(as CARLA runs on unreal engine)? or should be configured on GeForce graphics ?

**Comment by nagasanjay** [comment]
Did any one of you deleted the contents of `config` folder to load CarlaEditor fast ?

Because I did that and ran into similar issue. I solved it by downloading the config files again and putting it in config folder.

**Comment by germanros1987** [comment]
Issue closed due to lack of activity. Feel free to open a new issue if you still consider this is a critical matter.

**Comment by omkarshindetof** [comment]
Facing the same issue.

**Comment by omkarshinde22** [comment]
facing the same issue

**Comment by by-seong-me** [comment]
Before run python script, you should run server. In my case, something like this, 
$ ./CarlaUE4.sh /Game/Maps/Town01 -carla-server -benchmark -fps=15 -windowed -ResX=800 -ResY=600

And then, you can run python script in python example directory
$ python3 manual_control.py

**Comment by narayanaraju10** [comment]
Hello guys, recently while working with carla in python I used to get this error "RuntimeError: time-out of 2000ms while waiting for the simulator, make sure the simulator is ready and connected to 127.0.0.1:2000". 
This problem can be due to carla is not loading in 2000ms. in my case, i moved to higher GPU and RAM pc now its working fine and carla is loading with in 2000ms.

if you are stuck with same pc, kindly increase to pc to performance mode.

Best,
Narayana

**Comment by omkarshinde22** [comment]
Resolved this issue.
Way to solve this issue.
If you are using windows then please check your directx. Presently microsoft took off directx from its official website. But you can download it from third party.
Else you can increase the ram and rerun the Carla simulator.

**Comment by narayanaraju10** [comment]
change the time in   "client.set_timeout(2.0)" in the code to "client.set_timeout(200.0)"  # 200 depicts the loading time.

Best,
Narayana

**Comment by pinnintipraneethkumar** [comment]
> Hello guys, recently while working with carla in python I used to get this error "RuntimeError: time-out of 2000ms while waiting for the simulator, make sure the simulator is ready and connected to 127.0.0.1:2000".
> This problem can be due to carla is not loading in 2000ms. in my case, i moved to higher GPU and RAM pc now its working fine and carla is loading with in 2000ms.
> 
> if you are stuck with same pc, kindly increase to pc to performance mode.
> 
> Best,
> Narayana

Hii
May i know the what is your pc  specifications. 
Thanks.

**Comment by josephzhaoballer** [comment]
> Hello guys, recently while working with carla in python I used to get this error "RuntimeError: time-out of 2000ms while waiting for the simulator, make sure the simulator is ready and connected to 127.0.0.1:2000".
> This problem can be due to carla is not loading in 2000ms. in my case, i moved to higher GPU and RAM pc now its working fine and carla is loading with in 2000ms.
> 
> if you are stuck with same pc, kindly increase to pc to performance mode.
> 
> Best,
> Narayana

I changed the timeout to 5 seconds and it worked.

**Comment by NikhilKalghatgi** [comment]
> > Hello guys, recently while working with carla in python I used to get this error "RuntimeError: time-out of 2000ms while waiting for the simulator, make sure the simulator is ready and connected to 127.0.0.1:2000".
> > This problem can be due to carla is not loading in 2000ms. in my case, i moved to higher GPU and RAM pc now its working fine and carla is loading with in 2000ms.
> > if you are stuck with same pc, kindly increase to pc to performance mode.
> > Best,
> > Narayana
> 
> I changed the timeout to 5 seconds and it worked.

Can you please tell in which file you increased the timeout?

**Comment by josephzhaoballer** [comment]
client.set_timeout() look for this in which ever file you are running

**Comment by manju838** [comment]
Thanks @narayanaraju10 ,it worked for timeout as 5.0.Mine is a 2GB Graphics card and 2.0 time didn't work.Better the graphics,lesser timeout needed

**Comment by DonadiHarshith** [comment]
> Hi,
> 
> Could you tell us more about this, for example the steps you are taking to execute CARLA? Are you running it in Unreal Engine, or in the package version?

I am using package version can you help me

**Comment by shsrinivgit** [comment]
> Did you press the Play button (or alternatively Alt+P) after launching Unreal Engine, before running the scripts? Because I've only seen this error when you forget to do so.

THANKS. I JUST WASTED TIME REBUILDING THE CARLA

**Comment by e-Dylan** [comment]
tl;dr:
If it was open and working before you closed it, end the background process in task manager before you re-run the UE4 exe.

I just got this issue when I opened the CarlaUE4.exe unreal executable for carla, closed it, and tried to re-run it. I didn't realize the first one I had closed was still running an open process only visible in task manager. Once I found it in task manager (takes like 4gb memory but no cpu, not hard to find) and ended the task, api commands worked again after re-running the executable.

**Comment by gaooogao** [comment]
problem-solving methods:
I started the task manager and ended all related software.
then i restarted the emulator . Then ran the python script. 
this problem has been sloved.

if you have this problem, maybe you can try the methods. i hope it can work.

The process :
0.  when i use the version carla-0.9.13. 
1. I get the folder after I unzip the package
2. i executed CarlaUE4.exe. Then the it work.I can see Carla's map simulation interface.
3. Then I ran the Python script. i can control the car ,and change different cars.
4. Before I want to exit the Python script, however, I have already exited the emulation screen .
5. so I restarted the emulator and closed the Python script.
6. This error occurs when I start emulation again and run the Python script  .
7. Finally, I solved the problem through the above methods

**Comment by RocaPiedra** [comment]
> tl;dr: If it was open and working before you closed it, end the background process in task manager before you re-run the UE4 exe.
> 
> I just got this issue when I opened the CarlaUE4.exe unreal executable for carla, closed it, and tried to re-run it. I didn't realize the first one I had closed was still running an open process only visible in task manager. Once I found it in task manager (takes like 4gb memory but no cpu, not hard to find) and ended the task, api commands worked again after re-running the executable.

Hey, if you are using Windows and you randomly encounter this issue (for me it started happening right after performing some tests successfully, after restart it worked fine), this is the way to solve it!
Thanks for the info, Dylan

**Comment by Pratham544** [comment]
> tl;dr: If it was open and working before you closed it, end the background process in task manager before you re-run the UE4 exe.
> 
> I just got this issue when I opened the CarlaUE4.exe unreal executable for carla, closed it, and tried to re-run it. I didn't realize the first one I had closed was still running an open process only visible in task manager. Once I found it in task manager (takes like 4gb memory but no cpu, not hard to find) and ended the task, api commands worked again after re-running the executable.



> tl;dr: If it was open and working before you closed it, end the background process in task manager before you re-run the UE4 exe.
> 
> I just got this issue when I opened the CarlaUE4.exe unreal executable for carla, closed it, and tried to re-run it. I didn't realize the first one I had closed was still running an open process only visible in task manager. Once I found it in task manager (takes like 4gb memory but no cpu, not hard to find) and ended the task, api commands worked again after re-running the executable.

Thanks a lot, I tried increasing the timeout but it didn't work. Then i did this and it started working.

**Comment by RocaPiedra** [comment]
This issue happens in both Linux and Windows. In my experience, it mostly happens when you try to close the Carla Server from the application/script used. This is very annoying as I'm trying to create sort of an app that is self-managed (including Carla Server). I launch the server through a Python subprocess and end the subprocess when execution is finished but here is when the background process is an issue.
Does anybody know how to end the Carla Server cleanly?

**Comment by winterwindwang** [comment]
Same issue solved by increasing the timeout(2->10) of client.set_timeout(timeout)

**Comment by Kenneth71-P** [comment]
When I had the same problem, I entered task manager and ended CarlaUE4 app. Then it solved the problem.

**Comment by amirBgazand** [comment]
I had the same issue, and every time I get to this problem I just change the port to 3000, and amazingly, it works well.
just run the carlaUE4 by
CarlaUE4.exe -carla-rpc-port=3000
and in your code change the port to 3000 
client = carla.Client('127.0.0.1',3000)

**Comment by chiru42** [comment]
> tl;dr: If it was open and working before you closed it, end the background process in task manager before you re-run the UE4 exe.
> 
> I just got this issue when I opened the CarlaUE4.exe unreal executable for carla, closed it, and tried to re-run it. I didn't realize the first one I had closed was still running an open process only visible in task manager. Once I found it in task manager (takes like 4gb memory but no cpu, not hard to find) and ended the task, api commands worked again after re-running the executable.

This was happening to me. Thanks for mentioning about the hidden Background process. I end it from taskmanager and a fresh launch solved my issue.
